File: app/src/main/java/com/returnnotfound/stain/presentation/ui/main/MainPagerAdapter.java
package com.returnnotfound.stain.presentation.ui.main;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import com.returnnotfound.stain.presentation.ui.main.discover.DiscoverFragment;
import com.returnnotfound.stain.presentation.ui.main.love.LoveFragment;
import com.returnnotfound.stain.presentation.ui.main.rank.RankFragment;

public class MainPagerAdapter extends FragmentStateAdapter {
  private Fragment[] fragments = new Fragment[]{DiscoverFragment.newInstance(), RankFragment.newInstance(), LoveFragment.newInstance()};

  public MainPagerAdapter(@NonNull Fragment fragment) {
    super(fragment);
  }

  @NonNull
  @Override
  public Fragment createFragment(int position) {
    return fragments[position];
  }

  @Override
  public int getItemCount() {
    return fragments.length;
  }
}

File: app/src/main/java/com/returnnotfound/stain/presentation/ui/main/MainPagerAdapter.kt
package com.returnnotfound.stain.presentation.ui.main

import androidx.fragment.app.Fragment
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.returnnotfound.stain.presentation.ui.main.discover.DiscoverFragment.Companion.newInstance
import com.returnnotfound.stain.presentation.ui.main.love.LoveFragment
import com.returnnotfound.stain.presentation.ui.main.rank.RankFragment

class MainPagerAdapter(fragment: Fragment) :
  FragmentStateAdapter(fragment) {
  private val fragments =
    arrayOf<Fragment>(
      newInstance(),
      RankFragment.newInstance(),
      LoveFragment.newInstance()
    )

  override fun createFragment(position: Int): Fragment {
    return fragments[position]
  }

  override fun getItemCount(): Int {
    return fragments.size
  }
}